aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,170 @@
+
+### Compression with secure protocol
+Compressed responses over secure connections can be controlled with the `EnableForHttps` option, which is disabled by default. Using compression with dynamically generated pages can lead to security problems such as the [CRIME](https://en.wikipedia.org/wiki/CRIME_(security_exploit)) and [BREACH](https://en.wikipedia.org/wiki/BREACH_(security_exploit)) attacks.
+
+## Adding the Vary header
+When compressing responses based on the `Accept-Encoding` header, there are potentially multiple compressed versions of the response and an uncompressed version. In order to instruct client and proxy caches that multiple versions exist and should be stored, you should always supply a `Vary` header with an `Accept-Encoding` value. The sample application adds a `Vary` header using a method; however, the middleware will be upgraded soon to provide this feature ([Basic Middleware #187](https://github.com/aspnet/BasicMiddleware/issues/187)).
+
+[!code-csharp[Main](response-compression/sample/Startup.cs?name=snippet1)]
+
+## Disabling or removing IIS Dynamic Compression
+If you have an active IIS Dynamic Compression Module configured at the server level that you would like to disable for an application, you can do so with an addition to your *web.config* file. Either leave the module in place and deactivate it for dynamic compression or remove the module from the application.
+
+To merely deactivate dynamic compression module, add a `<urlCompression>` element to your *web.config* file. There is no need to include an attribute and value for `doStaticCompression="false"`, since the IIS Static Compression Module doesn't work with ASP.NET Core applications in a reverse-proxy setup.
+```xml
+<configuration>
+  <system.webServer>
+    <urlCompression doDynamicCompression="false"/>
+  </system.webServer>
+</configuration>
+```
+If you opt to remove the module via *web.config*, you must unlock it first. Click on the IIS server in the IIS Manager **Connections** sidebar. Open the **Modules** in the IIS area. Click on the **DynamicCompressionModule** in the list. In the **Action** panel on the right, click **Unlock**. At this point, you will be able to add the section shown below to your *web.config* file to remove the module from the application. Doing this won't affect your use of the module in other applications on the server.
+```xml
+<configuration>
+  <system.webServer> 
+    <modules> 
+      <remove name="DynamicCompressionModule" /> 
+    </modules> 
+  </system.webServer> 
+</configuration>
+```
+
+## Troubleshooting
+Use a tool like [Fiddler](http://www.telerik.com/fiddler), [Firebug](http://getfirebug.com/), or [Postman](https://www.getpostman.com/), all of which allow you to set the `Accept-Encoding` request header and study the response headers, size, and body. The Response Compression Middleware will compress responses that meet the following conditions:
```

Comment:
It might be more useful to move the tools section to the `Using defaults` section as a note where you show the response headers and response body.